test/typecheck/OWNERS

```diff
@@ -1,8 +1,8 @@
 # See the OWNERS docs at https://go.k8s.io/owners
 
 reviewers:
+  - BenTheElder
   - cblecker
-  - rmmh
```

Comment:
Note that I did NOT change approvers (below) in either file, though I am an approver of `test/` and `third_party` anyhow so perhaps that would make more sense. 
When I authored this I was thinking more about review ... @cblecker